### PR TITLE
Minor fixs

### PR DESCRIPTION
**Pull request by pifopi**

### CODE REVIEW

File: AutoMaxLair.py [hunk 1]
```diff
@@ -169,6 +169,7 @@ def battle(inst) -> str:
             inst.lives -= 1
             if inst.lives != 0:
                 inst.log('The lives counter was not 0.', 'WARNING')
+                inst.lives = 0
```
Comment (pifopi) [inline AutoMaxLair.py:172]: We need to put lives to 0 for two reason : 
- detect the lose and not award a win later on
- calculate the dynite ore gain

File: AutoMaxLair.py [hunk 2]
```diff
@@ -463,6 +464,10 @@ def select_pokemon(inst) -> str:
     if inst.num_caught == 0:
         inst.log('No Pokemon caught.')
         inst.push_buttons((None, 10), (b'b', 1))
+        inst.runs += 1
+        inst.reset_run()
+        inst.record_ore_reward()
```
Comment (pifopi) [inline AutoMaxLair.py:469]: it will award 0 ore but it will reset the reset counter to 0

File: AutoMaxLair.py [hunk 3]
```diff
@@ -546,7 +551,7 @@ def select_pokemon(inst) -> str:
     inst.push_buttons((b'b', 1.5), (b'b', 1.5))
     
     # Update statistics and reset stored information about the complete run.
-    inst.wins += 1 if inst.num_caught == 4 else 0
+    inst.wins += 1 if inst.lives != 0 else 0
```
Comment (pifopi) [inline AutoMaxLair.py:554]: we can win without catching 4 pokemon (when we're deciding not to catch the legendary in ball saver mode)

Comment (ercdndrs) [reply AutoMaxLair.py:554]: Nice catch!

**Pull request merged**